### PR TITLE
Fix `mypy` errors in `lorentz_transformation_four_vector.py`

### PR DESCRIPTION
**Pull request by tianyizheng02**
### Describe your change:

Fixed type hints to resolve `mypy` errors in `physics/lorentz_transformation_four_vector.py` (see #8070), added documentation, and refactored some things

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by CaedenPH — APPROVED** [review]
Very nice pull request!

**Review by ThomSullivan — APPROVED** [review]
The new formula formats are top notch!

**Comment by tianyizheng02** [comment]
@cclauss Could you take a look to see if it's good to merge?

File: physics/lorentz_transformation_four_vector.py [hunk 1]
```diff
@@ -1,119 +1,111 @@
...
     ValueError: Speed must be greater than 1!
     """
     if velocity > c:
-        raise ValueError("Speed must not exceed Light Speed 299,792,458 [m/s]!")
-
-    # Usually the speed u should be much higher than 1 (c order of magnitude)
+        raise ValueError("Speed must not exceed light speed 299,792,458 [m/s]!")
     elif velocity < 1:
+        # Usually the speed should be much higher than 1 (c order of magnitude)
         raise ValueError("Speed must be greater than 1!")
```
Comment (cclauss) [inline physics/lorentz_transformation_four_vector.py:59]:
```suggestion
        raise ValueError("Speed must be greater than or equal to 1!")
```

Comment (tianyizheng02) [reply physics/lorentz_transformation_four_vector.py]: Updated the error message

**Review by cclauss — APPROVED** [review]

**Pull request merged**